bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/RuleRegistryImpl.java

```diff
@@ -0,0 +1,99 @@
+/*******************************************************************************
+ * Copyright (c) 1997, 2015 by ProSyst Software GmbH
+ * http://www.prosyst.com
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    ProSyst Software GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.smarthome.automation.core;
+
+import java.util.Collection;
+import java.util.Set;
+
+import org.eclipse.smarthome.automation.Rule;
+import org.eclipse.smarthome.automation.RuleRegistry;
+import org.eclipse.smarthome.core.common.registry.AbstractRegistry;
+
+/**
+ * @author Yordan Mihaylov - Initial Contribution
+ */
+public class RuleRegistryImpl extends AbstractRegistry<Rule, String> implements RuleRegistry {
+
+    private RuleManager ruleManager;
+
+    public RuleRegistryImpl(RuleManager ruleManager, RuleProvider rp) {
+        this.ruleManager = ruleManager;
+        this.addProvider(rp);
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see org.eclipse.smarthome.core.common.registry.Registry#get(java.lang.Object)
+     */
+    @Override
+    public Rule get(String key) {
```

Comment:
only methods of RuleRegistry should be implemented here, the rest is already taken care of by the AbstractRegistry.